java/vector/src/main/java/org/apache/arrow/vector/BaseNullableVariableWidthVector.java

```diff
@@ -761,4 +741,57 @@ protected final void handleSafe(int index, int dataLength) {
          reallocValueBuffer();
       }
    }
+
+
+   /******************************************************************
+    *                                                                *
+    *                helper methods currently                        *
+    *                used by JsonFileReader and                      *
+    *                JsonFileWriter                                  *
+    *                                                                *
+    ******************************************************************/
+
+
+   public static ArrowBuf setSafeJsonHelper(ArrowBuf data, ArrowBuf offset,
```

Comment:
@siddharthteotia can you add docs to explain this function? Also does this have to be public API?

Reply:
Done.